types/store/rhs.ts

```diff
@@ -23,6 +23,10 @@ export type PostDraft = {
     uploadsInProgress: string[];
     props?: any;
     caretPosition?: number;
+    channelId: string;
+    createAt: number;
+    updateAt: number;
+    show?: boolean;
```

Comment:
What's this `show` for again? If it's to differentiate empty drafts from ones that we want to see in the draft view, we could make a helper to check if any of `message`, `fileInfos`, and `uploadsInProgress` are set, and if they are, then we show the draft rather than saving a separate field for it

Reply:
It's for a UX requirement. A draft will be displayed in global draft when you focused out of the input.

Reply:
Ah, I see. Could we perhaps have that derived by having the Global Drafts menu link be aware of the current channel ID and selected post ID? I'd be worried that storing it will somehow cause the stored value to become out of sync with the rest of the UI or cause some weirdness when you have multiple tabs open viewing different channels

Reply:
hm I don't know how it would work.
Let's say you have no drafts and visiting channel A and started writing a message,  
the link "knows" you're on channel A and that the draft belongs there,
so it will be hidden.
Now you visit channel B and the link shows itself, 
but what if you go back to channel A having only that one draft?

Reply:
That's a good point. I can't think of a way to derive that unless we were to do something with not putting the draft in redux until you blur the textbox, but I'd be worried about that causing the draft to sometimes not save at all.

The only other concern that I'd have with this then is if we're persisting the value of `show` to storage. Is it possible for the user to close the tab without triggering the blur so that we end up with a draft saved with a show value of false? I suppose we could use the `REHYDRATE` action to ensure its always set to `true` when rehydrating if it does become an issue though.

Do you think that's a possiblity? If not, I'd be good leaving this as-is

Reply:
^ Bump on this since I realize GitHub doesn't make it immediately obvious when someone responds to a thread without flagging it as a whole review

Reply:
> Is it possible for the user to close the tab without triggering the blur so that we end up with a draft saved with a show value of false?

@hmhealey I've tried closing, and refreshing the tab via shortcut blur is triggered, can't think of any other way the user would not blur the input and close.